rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/RootsStore.scala

```diff
@@ -0,0 +1,50 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+
+import cats.implicits._
+import cats.effect.Sync
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.shared.AttemptOps._
+import coop.rchain.shared.ByteVectorOps._
+import scodec.bits.{BitVector, ByteVector}
+
+trait RootsStore[F[_]] {
+  def currentRoot(): F[Option[Blake2b256Hash]]
+  def validateRoot(key: Blake2b256Hash): F[Option[Blake2b256Hash]]
+  def recordRoot(key: Blake2b256Hash): F[Unit]
+
+  def close(): F[Unit]
+}
+
+object RootsStoreInstances {
+  def rootsStore[F[_]: Sync](store: Store[F]) = new RootsStore[F] {
+    val tag: ByteBuffer = ByteVector("root".getBytes(StandardCharsets.UTF_8)).toDirectByteBuffer
+    val currentRootName: ByteBuffer =
+      ByteVector("current-root".getBytes(StandardCharsets.UTF_8)).toDirectByteBuffer
+
+    override def currentRoot(): F[Option[Blake2b256Hash]] =
+      store
+        .get(currentRootName)
+        .map(_.map(bytes => Blake2b256Hash.codecBlake2b256Hash.decode(BitVector(bytes)).get.value))
+
+    override def validateRoot(key: Blake2b256Hash): F[Option[Blake2b256Hash]] = {
+      val bytes = Blake2b256Hash.codecBlake2b256Hash.encode(key).get.toByteVector.toDirectByteBuffer
+      for {
+        bytesMaybe <- store.get(bytes)
+        result     <- bytesMaybe.traverse(_ => store.put(currentRootName, bytes).map(_ => key))
```

Comment:
Here you ignore the bytes from bytesMaybe. Is that ok?

I'm pretty confused by the fact that you are doing the store.get and then ignore its output

Reply:
if the root exists - set it as "current". otherwise -> none